frontend/src/hooks/api/actions/useApi/useApi.ts

```diff
@@ -104,6 +110,28 @@ const useAPI = ({
                 }
             }
 
+            if (res.status === UNAVAILABLE) {
+                if (handleUnavailable) {
+                    return handleUnavailable(setErrors, res, requestId);
+                } else {
+                    setErrors((prev) => ({
+                        ...prev,
+                        unavailable: 'This operation is unavailable',
+                    }));
+                }
+
+                if (propagateErrors) {
+                    if (maintenanceMode) {
+                        throw new Error(
+                            'Operation unavailable: Maintenance mode is enabled',
```

Comment:
Should we set this message in the backend instead? If you're using an API I believe you'd like to know as well...

Reply:
Fair point. How about this? https://github.com/Unleash/unleash/pull/5018/commits/d9ea8a6c4ffde382e39195a6a2adfe24c56d6544

Reply:
Nice, should we just get the message sent from the backend?

Reply:
I mean, that way we only change it on one side

Reply:
How about this then? https://github.com/Unleash/unleash/pull/5018/commits/a1323c994889541fd8946f4d0fddd5cbbd385bf4